### PR TITLE
✨ wizard: anomalist (first draft)

### PR DESCRIPTION
**Pull request by lucasrodes**
First implementation of: https://github.com/owid/etl/issues/3340

- Re-organisation of grapher db/s3 util functions.
- Draft of anomalist page in Wizard.

### CODE REVIEW

**Comment by lucasrodes** [comment]
implemented the foundations of the app.

next tasks:

- create plot interface (entity→ country)
- add button to generate anomaly list, per indicator
    - can we have sth to list those we want to create lists for? checkboxes per container; maybe too long. multiselect at first + individual buttons? ー second option sounds better
    - individual button action should be kept in fragment ー avoid whole page rerun
- think plot for upgrade mode
- export anomalies somewhere, maybe indicator metadata?

**Comment by lucasrodes** [comment]
Currently working on streaming the output from OpenAI and render streamlit objects.

Relevant links:
- Structured output: https://platform.openai.com/docs/guides/structured-outputs/supported-schemas
- Streaming structured output: https://community.openai.com/t/streaming-using-structured-outputs/925799/8
- intro to structured output: https://cookbook.openai.com/examples/structured_outputs_intro

File: etl/grapher_model.py [hunk 1]
```diff
@@ -1251,6 +1277,24 @@ def override_yaml_path(self) -> Path:
         """Return path to indicator YAML file."""
         return self.step_path.with_suffix(".meta.override.yml")
 
+    def get_data(self, session: Optional[Session] = None) -> pd.DataFrame:
```
Comment (Marigold) [inline etl/grapher_model.py:1280]: There's already a function `variable_data_df_from_s3` for reading data from S3, you can reuse it.

**Review by Marigold — APPROVED** [review]
It's hard to review such a massive PR, but it looks good after skimming through it. Don't forget to squash it.

**Pull request merged**